google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreImpl.java

```diff
@@ -131,29 +136,41 @@ public T call() throws DatastoreException {
   @Override
   public <T> T runInTransaction(final TransactionCallable<T> callable) {
     final DatastoreImpl self = this;
+    Span span = traceUtil.startSpan(TraceUtil.SPAN_NAME_TRANSACTION);
+    Scope scope = traceUtil.getTracer().withSpan(span);
     try {
       return RetryHelper.runWithRetries(
           new ReadWriteTransactionCallable<T>(self, callable, null),
           retrySettings,
           TRANSACTION_EXCEPTION_HANDLER,
           getOptions().getClock());
     } catch (RetryHelperException e) {
+      span.setStatus(Status.UNKNOWN.withDescription(e.getMessage()));
       throw DatastoreException.translateAndThrow(e);
+    } finally {
+      scope.close();
+      span.end(TraceUtil.END_SPAN_OPTIONS);
     }
   }
 
   @Override
   public <T> T runInTransaction(
       final TransactionCallable<T> callable, TransactionOptions transactionOptions) {
     final DatastoreImpl self = this;
+    Span span = traceUtil.startSpan(TraceUtil.SPAN_NAME_TRANSACTION);
+    Scope scope = traceUtil.getTracer().withSpan(span);
```

Comment:
You can use try with resources for the scope

Reply:
Done